### PR TITLE
Preparing 1.4.8

### PR DESCRIPTION
**Pull request by Didou09**
See release notes for details

Main changes:
--------------------
* Optimized volume sampling from subdomain and indices in toroidal geometry
* Debugged / optimized IMAS compatibility
* More flexible bah commands
* New geometries: COMPASS and TCV
* Minor debugging, better doc, better error messages

### CODE REVIEW

**Review by lasofivec — APPROVED** [review]

**Review by Didou09 — COMMENTED** [review]
Thanks for taking care of this

**Comment by lasofivec** [comment]
@Didou09 

Not sure if the following line should be commented ? I am having trouble seeing when was this commented

https://github.com/ToFuProject/tofu/blob/d9fc680b4e6ffdb017ec8578c5ec1f78b3d91c15/tofu/tests/tests01_geom/tests03_core.py#L377

**Comment by Didou09** [comment]
> @Didou09
> 
> Not sure if the following line should be commented ? I am having trouble seeing when was this commented
> 
> https://github.com/ToFuProject/tofu/blob/d9fc680b4e6ffdb017ec8578c5ec1f78b3d91c15/tofu/tests/tests01_geom/tests03_core.py#L377

Hum, leave it as is for now, I'll look into it later but I think it is fine

**Comment by lasofivec** [comment]
I think the PEP8 comments are all solved, for some reason it hasnt been updated

**Review by lasofivec — APPROVED** [review]

File: tofu/tests/tests01_geom/tests03_core.py [hunk 1]
```diff
@@ -421,12 +421,13 @@ def test15_get_sampleV(self):
                             pts0, ind0 = out[0], out[2]
                         except Exception as err:
                             msg = (str(err) +
-                                   "\nFailed for {0}_{1}_{2}\n".format(typ, c, n)
+                                   "\nFailed for {0}_{1}_{2}\n".format(typ,
+                                                                       c, n)
```
Comment (lasofivec) [inline tofu/tests/tests01_geom/tests03_core.py:425]: Here for example the updated version, but black isnt showing them :shrug:

Comment (Didou09) [reply tofu/tests/tests01_geom/tests03_core.py:425]: ok, fine for me , thanks !

**Review by lasofivec — APPROVED** [review]

**Comment by lasofivec** [comment]
👍

**Comment by Didou09** [comment]
I waited for all Travis builds to pass, they finished around 2:00 am this night, so I merged this morning, 1.4.8 is on the way :-)

**Pull request merged**